internal/datasystem/store.go

```diff
@@ -0,0 +1,252 @@
+package datasystem
+
+import (
+	"sync"
+
+	"github.com/launchdarkly/go-server-sdk/v7/internal/memorystorev2"
+
+	"github.com/launchdarkly/go-server-sdk/v7/internal/fdv2proto"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldlog"
+	"github.com/launchdarkly/go-server-sdk/v7/interfaces"
+	"github.com/launchdarkly/go-server-sdk/v7/subsystems"
+	"github.com/launchdarkly/go-server-sdk/v7/subsystems/ldstoretypes"
+)
+
+// Store is a dual-mode persistent/in-memory store that data queries from the evaluation
+// algorithm.
+//
+// At any given moment one of two stores is active: in-memory, or persistent. This doesn't preclude a caller
+// from holding on to a reference to the persistent store even when we swap to the in-memory store.
+//
+// Once the in-memory store has data (either from initializers or a synchronizer), the persistent
+// store is no longer read from. From that point forward, calls to Get will serve data from the memory
+// store.
+//
+// One motivation behind using persistent stores is to offer a way to immediately start evaluating
+// flags before a connection is made to LD (or even in the moment before an initializer has run.)
```

Comment:
Sorry for the nitpick.

```suggestion
// flags before a connection is made to LD (or even in the moment before an initializer has run).
```